src/js/player.js

```diff
@@ -280,16 +266,24 @@ class Player extends Component {
     // Turn off API access because we're loading a new tech that might load asynchronously
     this.isReady_ = false;
 
-    var techReady = function(){
-      this.player_.triggerReady();
-    };
+    var techReady = Lib.bind(this, function() {
+      this.triggerReady();
+    });
 
     // Grab tech-specific options from player options and add source and parent element to use.
-    var techOptions = Lib.obj.merge({ 'source': source, 'parentEl': this.el_ }, this.options_[techName.toLowerCase()]);
+    var techOptions = Lib.obj.merge({
+      'source': source,
+      'playerId': this.id(),
+      'textTracks': this.textTracks_
+    }, this.options_[techName.toLowerCase()]);
```

Comment:
if tracks are added manually after the config and then the tech changes, would this pick up these changes?